src/data.rs

```diff
@@ -18,9 +18,7 @@ pub(crate) struct NodeData {
     pub(crate) needs_measure: bool,
 
     /// The primary cached results of the layout computation
-    pub(crate) main_size_layout_cache: Option<Cache>,
-    /// Secondary cached results of the layout computation
-    pub(crate) other_layout_cache: Option<Cache>,
+    pub(crate) size_cache: [Option<Cache>; 4],
```

Comment:
Just so I understand this correctly: Instead of having an explicit primary and secondary cache, we now just have a cache with 4 "slots"?

Reply:
Yes, correct. Effectively we did have a cache with 2 slots, and now we have a cache with 4 slots (which has been turned into an array to make this more obvious (and also to make it easy to change the size in future should we wish to).